src/app/core-ui/material/mat-otpgroup-select-search/mat-otpgroup-select-search.component.ts

```diff
@@ -0,0 +1,67 @@
+import { Component, Input, Output, EventEmitter, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { Observable } from 'rxjs';
+import { startWith, map } from 'rxjs/operators';
+
+@Component({
+  selector: 'mat-otpgroup-select-search',
+  templateUrl: './mat-otpgroup-select-search.component.html',
+  styleUrls: ['./mat-otpgroup-select-search.component.scss']
+})
+export class MatOtpGroupSelectSearchComponent implements OnInit {
+  stateForm: FormGroup = this.fb.group({
+    stateGroup: '',
+  });
+  @Input() defaultOption: string = '';
+  @Input() options: any[] = [];
+  @Output() change: EventEmitter<any> = new EventEmitter<any>();
+  @Input() placeHolder: string = '';
+  @Input() isRequired: boolean = false;
+
+  stateGroupOptions: Observable<any[]>;
+
+  constructor(private fb: FormBuilder) { }
+
+  ngOnInit() {
+    this.stateGroupOptions = this.stateForm.get('stateGroup')!.valueChanges
+      .pipe(
+        startWith(''),
+        map(value => {
+          return this._filterGroup(value)
+        })
+      );
+  }
+
+  private _filterGroup(value: string): any[] {
+    if (!this.options || !this.options.length) {
+      return []
+    }
+
+    if (value) {
+      return this.options
+        .map(group => ({ name: group.name, subCategoryList: this._filter(group.subCategoryList, value) }))
```

Comment:
This has to be a completely generic piece of code, it shouldn't have any references to subCategoryList please change the name.